core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -397,8 +398,7 @@ protected Scope visitInsert(Insert insert, Optional<Scope> scope)
 
             // analyze the query that creates the data
             Scope queryScope = analyze(insert.getQuery(), createScope(scope));
-
-            analysis.setUpdateType("INSERT", targetTable, Optional.empty());
+            analysis.setUpdateType("INSERT", targetTable, Optional.empty(), ImmutableList.of());
```

Comment:
remove? you are doing this below anyway and this one here is incorrect as it has empty list of columns

Reply:
If the query fails like if the table doesn't exist then `updateType` is not set. So do we skip setting on `updateType` on failed queries ?

Reply:
please add a comment there

Reply:
please also add a tests for such query failure